src/driver_ros1.cpp

```diff
@@ -41,17 +41,21 @@ DriverROS1::DriverROS1(ros::NodeHandle & nh) : nh_(nh)
 
   if (wrapper_->getSyncMode() == "primary") {
     // defer starting the primary until the secondary is up
-    const std::string topic = "ready";
-    ROS_INFO_STREAM("waiting for ready message from secondary on topic \"" << topic << "\"");
-    secondaryReadySub_ = nh_.subscribe(topic, 1, &DriverROS1::secondaryReadyCallback, this);
-  } else {
-    if (wrapper_->getSyncMode() == "secondary") {
-      secondaryReadyPub_ = nh_.advertise<std_msgs::Header>("ready", 1);
-    }
-    if (!start()) {
-      ROS_ERROR("startup failed!");
-      throw std::runtime_error("startup of CameraDriver node failed!");
+    ros::ServiceClient client = nh_.serviceClient<Trigger>("ready");
+    Trigger trig;
+    while (!client.call(trig)) {
+      ROS_INFO_STREAM("waiting for secondary to come up...");
+      ros::Duration(0, 1000000000).sleep();  // sleep for a second
     }
+    ROS_INFO_STREAM("secondary is up: " << trig.response.message);
+    start();  // only now can this be started
+  } else if (wrapper_->getSyncMode() == "secondary") {
+    //
```

Comment:
Looks like an empty comment.

Reply:
Also looks like a bug :(.
The effect is so small that I didn't catch it when testing.